main.py

```diff
@@ -59,6 +80,50 @@ def configure_confluent_kafka_consumer(event, args):
     message = json.loads(event['Records'][0]['Sns']['Message'])
     logger.debug(message)
 
+    private_ip = message['detail']['containers'][0]['networkInterfaces'][0]['privateIpv4Address']
+    logger.debug(private_ip)
+
+    payload = {
+        "config": {
+            "connector.name": "s3-sink",
+            "connector.class": "io.confluent.connect.s3.S3SinkConnector",
+            "tasks.max": args.tasks_max,
+            "topics": args.topics,
+            "flush.size": args.flush_size,
+            "s3.region": args.aws_region,
+            "s3.bucket.name": args.s3_bucket_name,
+            "storage.class": "io.confluent.connect.s3.storage.S3Storage",
+            "format.class": "io.confluent.connect.s3.format.avro.AvroFormat",
+            "schema.generator.class": "io.confluent.connect.storage.hive.schema.DefaultSchemaGenerator",
+            "partitioner.class": "io.confluent.connect.storage.partitioner.DefaultPartitioner",
+        }
+    }
+    response = requests.get(f"http://{private_ip}:8083/connectors")
+    existing_connector = response.content.decode("utf-8")
+    logger.debug(existing_connector)
+    print (existing_connector)
+
+    # existing_connector = ['s3-sink', 'other-connector']
+
+    if existing_connector in existing_connector:
+        logger.debug("update connector [PUT]")
+
+        response = requests.put(f"http://{private_ip}:8083/connectors", json=payload)
```

Comment:
url and payload are different for put.
https://docs.confluent.io/current/connect/references/restapi.html#put--connectors-(string-name)-config